### ISSUE TITLE
Linux save game location

### ISSUE THREAD

**Issue by ghost** [dukope/BugsPlease#7]
Seems more appropriate to actually put this in the bug tracker.

If you're intending on using autocloud for Linux same as for the other platforms, you will have to change the location to $XDG_DATA_HOME instead. Currently, Steam only supports the following two Linux specific paths (besides AppInstall and SteamCloud directories, which work on all systems).

LinuxXdgDataHome (uses $XDG_DATA_HOME or ~/.local/share/ if unavailable) is the only envvar that's supported by Steam Cloud. _Technically_ LinuxHome could do the job, but that would not account for anyone having $XDG_CONFIG_HOME set to anything other than ~/.config/

This would also be more in line with the base directory specification[1].

On a personal note, I wouldn't mind more typical directory name. What about just saving to $XDG_DATA_HOME/3909/PapersPlease/ ? :)

[1] http://standards.freedesktop.org/basedir-spec/basedir-spec-latest.html

**Comment by dukope** [comment]
I'll look into this. Right now it's just using the OpenFL-chosen directory for saves, but it sounds like that's not ideal. I can probably override it on my end and let them know about the issue.

**Comment by dukope** [comment]
This is the OpenFL/lime logic:

``` C++
std::string gPackage = "org.haxe.lime";
...
    else if (inDir == DIR_STORAGE)
    {
        if (getenv("XDG_CONFIG_HOME") != NULL)
        {
            outDir = std::string(getenv ("XDG_CONFIG_HOME")) + "/" + gPackage;
        }
        else
        {
            outDir = std::string(getenv ("HOME")) + "/.config/" + gPackage;
        }
    }
```

So yeah, I'll override that manually using XDG_DATA_HOME and the actual company/app name

(Note that Steam's cross-platform cloud saves have caused nothing but problems so I probably won't enable them for Linux+Win+OSX any time soon)

**Comment by jgranick** [comment]
"$XDG_DATA_HOME defines the base directory relative to which user specific data files should be stored. If $XDG_DATA_HOME is either not set or empty, a default equal to $HOME/.local/share should be used.

$XDG_CONFIG_HOME defines the base directory relative to which user specific configuration files should be stored. If $XDG_CONFIG_HOME is either not set or empty, a default equal to $HOME/.config should be used."

It seems we are using the correct logic for application configuration files, but I suppose the data saved by SharedObject is more likely to be considered user data, not configuration?

Open to suggestions for improvement here, not sure if it would make sense to support both, or to just switch to "data" instead of "config" instead

**Comment by ghost** [comment]
Yes, you're creating picking the directory correctly. The best approach is supporting both, however currently it is typical of games to just use one, mostly $XDG_DATA_HOME. There is a great ideal when both are used - one can share save data between systems without sharing configuration. While that doesn't matter for this game, it makes a difference in other cases - so supporting and correctly using both would be nice. But I think it'd be good enough to just switch to data in place of config.

I consider saves to be data rather than configuration, but that is in the end a matter of opinion.

**Comment by Tailszefox** [comment]
I think $XDG_DATA_HOME is a good place to store it. I don't have it set, so it defaults to ~/.local/share, which is where most games I played store their savefiles: I see FTL, Fez, Proteus, Runner 2...Most of these games also have a directory in $XDG_CONFIG_HOME where the configuration files are stored, so the difference that @soeb is talking about is implemented that way: savefiles on one side, configuration files on the other.

As long as it's not lying around in its own hidden directory at the root of the $HOME directory, it's already good, I think. But just keeping $XDG_DATA_HOME seems to make the most sense and be the most consistent with other games.

**Comment by dukope** [comment]
Josh, I say just use $XDG_DATA_HOME for saves.

I also have a configuration file for the game (boot.xml) that may be better placed in $XDG_CONFIG_HOME. But that's something I'll deal with separately (most likely by waiving it).

**Comment by dukope** [comment]
Fixed in 1.1.56
